[WGSL] Add missing Texture entries to the AST dumping

tadeuzagallo opened the pull request:
#### 0d25214798937779c36c96b86b359221ca451deb
<pre>
[WGSL] Add missing Texture entries to the AST dumping
<a href="https://bugs.webkit.org/show_bug.cgi?id=254421">https://bugs.webkit.org/show_bug.cgi?id=254421</a>
rdar://107186502

Reviewed by Myles C. Maxfield.

In <a href="https://commits.webkit.org/262063@main">https://commits.webkit.org/262063@main</a> we added support for Textures in the compiler,
which required adding extra entries to ParameterizedTypeName::Base, but I missed adding
the same entries to ASTStringDuper. To avoid running into the same problem in the future
I moved the toString logic next to the enum definition and used an explicit switch, that
way the compiler will warns us about unhandled entries.

* Source/WebGPU/WGSL/AST/ASTStringDumper.cpp:
(WGSL::AST::StringDumper::visit):
* Source/WebGPU/WGSL/AST/ASTTypeName.h:
(WGSL::AST::ParameterizedTypeName::baseToString):

Canonical link: <a href="https://commits.webkit.org/262146@main">https://commits.webkit.org/262146@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25c8a008a7f1f1180c97adc1355fc0ec72946b12

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/746 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/768 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/795 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/984 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/665 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/817 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/851 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/891 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/753 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/713 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/713 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/941 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/776 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/698 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/722 "6 flakes 139 failures") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/678 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/719 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/1689 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/709 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/694 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/662 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/704 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/169 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/719 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WGSL/AST/ASTTypeName.h:
```diff
@@ -177,6 +177,58 @@ class ParameterizedTypeName : public TypeName {
 
+    static String baseToString(Base base)
+    {
+        switch (base) {
```
Why not just index an array of Strings? To catch mismatch between enum and strings?

tadeuzagallo replied:
Exactly. Although I guess we could have an assertion that the size of the array is the same as the number of entries in Base, but I thought this was simpler.

djg commented on Source/WebGPU/WGSL/AST/ASTTypeName.h:
```diff
@@ -177,6 +177,58 @@ class ParameterizedTypeName : public TypeName {
         return std::nullopt;
     }
 
+    static String baseToString(Base base)
```
Why String instead of ASCIILiteral?

tadeuzagallo replied:
No particular reason, just because I knew it was going to be used as a String, but it might be better to make it ASCIILiteral, I'll change.

djg reviewed (approved):
:shipit:

Just minor questions, not even nits.

eocanha committed:
[MSE][GStreamer] Missing support for aborts not followed by an initialization segment
https://bugs.webkit.org/show_bug.cgi?id=228820

This patch performs a flush on the AppendPipeline on SourceBuffer::abort().
Such action drains the AppendPipeline but leaves the demuxer still configured with
the context provided by the last init segment. This is in compliance with the spec,
which mandates that there's no need to append an init segment after an abort,
because the last one should be reused.

For this patch to work, the following GStreamer merge requests, scheduled to be
landed on 1.23.1, must be present:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4101.
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4199.

If the GStreamer version is lower than the required one, the old reset technique,
consisting of setting the pipeline to READY and then PLAYING, is applied.

In the current circumstances, the layout tests still don't pass, so this patch
isn't unskipping them yet. media-mp4-h264-partial-abort.html should be unskipped
when we start using a GStreamer version that includes the MRs.
media-webm-opus-partial-abort.html should be unskipped when
https://github.com/WebKit/WebKit/pull/11807 lands.

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/issues/1016

Reviewed by Alicia Boya Garcia.

* Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp:
(WebCore::AppendPipeline::resetParserState): If supported by GStreamer, perform a flush instead of setting the pipeline state to READY and then again to PLAYING.

Canonical link: https://commits.webkit.org/262144@main

cathiechen committed:
[Resize-observer] The `getBoundingBox` of a non-empty SVG element is 0x0
https://bugs.webkit.org/show_bug.cgi?id=252903

Reviewed by Oriol Brufau and Nikolas Zimmermann.

Per [1], the specification updates that for SVG elements that do not have associated CSS layout boxes define bounding box
instead of a content box. Previously, all svg elements are using the bounding box. This patch adds Element::getBoundingBoxForSVGElement()
and reuses shouldObtainBoundsFromSVGModel to get the bounding box only for svg elements without associated CSS layout boxes.

* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/contain-intrinsic-size/auto-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/svg-with-css-box-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/svg-with-css-box-002-expected.txt:
* Source/WebCore/dom/Element.cpp:
(WebCore::shouldObtainBoundsFromSVGModel):
* Source/WebCore/page/ResizeObservation.cpp:
(WebCore::ResizeObservation::computeObservedSizes const):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::hasAssociatedSVGLayoutBox const):
* Source/WebCore/svg/SVGElement.h:

Canonical link: https://commits.webkit.org/262145@main

tadeuzagallo committed:
[WGSL] Add missing Texture entries to the AST dumping
https://bugs.webkit.org/show_bug.cgi?id=254421
rdar://107186502

Reviewed by Myles C. Maxfield.

In https://commits.webkit.org/262063@main we added support for Textures in the compiler,
which required adding extra entries to ParameterizedTypeName::Base, but I missed adding
the same entries to ASTStringDuper. To avoid running into the same problem in the future
I moved the toString logic next to the enum definition and used an explicit switch, that
way the compiler will warns us about unhandled entries.

* Source/WebGPU/WGSL/AST/ASTStringDumper.cpp:
(WGSL::AST::StringDumper::visit):
* Source/WebGPU/WGSL/AST/ASTTypeName.h:
(WGSL::AST::ParameterizedTypeName::baseToString):

Canonical link: https://commits.webkit.org/262146@main
